src/coreclr/vm/frozenobjectheap.h

```diff
@@ -28,7 +28,15 @@ class FrozenObjectHeapManager
 public:
     FrozenObjectHeapManager();
     Object* TryAllocateObject(PTR_MethodTable type, size_t objectSize, bool publish = true);
-
+    FrozenObjectSegment** GetSegments(unsigned* count)
+    {
+        *count = m_FrozenSegments.GetCount();
+        return m_FrozenSegments.GetElements();
+    }
+    Crst *GetCrst()
```

Comment:
Alternative design: expose `Enumerate(Action<object> callback)`-like API that will not expose the lock.

Reply:
thanks, addressed